addons/website/controllers/form.py

```diff
@@ -327,3 +329,90 @@ def insert_attachment(self, model_sudo, id_record, files):
             # attach the custom binary field files on the attachment_ids field.
             for attachment_id_id in orphan_attachment_ids:
                 record.attachment_ids = [(4, attachment_id_id)]
+
+class CustomAuthSignup(AuthSignupHome):
+    @http.route("/web/signup", type="http")
+    def web_auth_signup(self, *args, **kwargs):
+        response = super().web_auth_signup(*args, **kwargs)
+
+        if "error" in response.qcontext:
+            return request.make_response(
+                json.dumps({"error": response.qcontext["error"]}),
+                headers=[("Content-Type", "application/json")]
+            )
+
+        login = kwargs.get("login")
+        if not login or response.status_code not in range(200, 308):
+            return response
+
+        users = self.env["res.users"].sudo().search([("login", "=", login)])
+        if not users:
+            return response
+
+        partner = users.partner_id.sudo()
+
+        existing_fields = set(self.env["res.users"]._fields.keys())
+        remaining_fields = {
+            key: value for key, value in kwargs.items()
+            if key not in existing_fields and key != "confirm_password"
+        }
+
+        # Separate text fields and file attachments
+        attachments, text_fields = [], {}
+
+        for key, value in remaining_fields.items():
+            if isinstance(value, werkzeug.datastructures.FileStorage):
+                attachment = self.env["ir.attachment"].sudo().create({
+                    "name": value.filename,
+                    "datas": base64.b64encode(value.read()).decode("utf-8"),
+                    "res_model": "res.partner",
+                    "res_id": partner.id,
+                    "mimetype": value.content_type,
+                })
+                attachments.append(attachment.id)
+            else:
+                text_fields[key] = value
+
+        # Prepare message body efficiently
+        message_body_parts = []
+
+        if text_fields:
+            message_body_parts.append("<p><strong>Other Information:</strong></p><ul>")
+            message_body_parts.extend(
+                f"<li><strong>{key}:</strong> {value}</li>" for key, value in text_fields.items()
+            )
+            message_body_parts.append("</ul>")
+
+        if attachments:
+            message_body_parts.append("<p><strong>📎 Attachment Files</strong></p>")
+
+        # Log message in chatter
+        if message_body_parts or attachments:
+            partner._message_log(
+                body=Markup(''.join(message_body_parts)),
```

Comment:
i would have sanitized the data that we are getting kwargs before assigning it to body.